iterative/dvc
# [WIP] testing subcommands with pluggy

#### How to test:
The `plugins` directory contains a reference `dvc hello` subcommand. 
1. check out this PR, and chdir to root of the repo.
2. `pip install -e "."` if required.
2. Invoke dvc as:
 ```bash
  dvc --plugins plugins [hello]
  ```

You can add more files to this directory while testing, or in a new separate directory. The only thing that you need to implement in the file is `register_command()` as done in [`hello.py`](https://github.com/iterative/dvc/blob/706455cc324a2743002bc3a2c7b8990b23497887/plugins/hello.py).

If you want to test out with a real project, you can also try by creating a simple python project. Here's a simple script to run previous `hello` example as a package:
```console
mkdir dvc-hello
cp plugins/hello.py dvc-hello
pushd dvc-hello
cat << EOF > setup.py
from setuptools import setup
setup(name="dvc-hello", entry_points={"dvc": ["hello = hello"]}, py_modules=["hello"])
EOF
pip install -e "."
popd
dvc --help
dvc hello
```
